app/assets/javascripts/velumMinionPoller.js

```diff
@@ -0,0 +1,123 @@
+        // passed in so we can ensure the value of it is truly undefined. In ES5, undefined
+        // can no longer be modified.
+
+        // window and document are passed through as local variables rather than global
+        // as this (slightly) quickens the resolution process and can be more efficiently
+        // minified (especially when both are regularly referenced in your plugin).
+
+        // Create the defaults once
+        var pluginName = "velumMinionPoller",
+            defaults = {
+                timeoutInterval: 5000,
+                nodesUrl: "/nodes"
+            };
+
+        // The actual plugin constructor
+        function Plugin ( element, options ) {
+            this.element = element;
+
+            // jQuery has an extend method which merges the contents of two or
+            // more objects, storing the result in the first object. The first object
+            // is generally empty as we don't want to alter the default options for
+            // future instances of the plugin
+            this.settings = $.extend( {}, defaults, options );
+            this._defaults = defaults;
+            this._name = pluginName;
+            this.init();
+        }
+
+        // Avoid Plugin.prototype conflicts
+        $.extend( Plugin.prototype, {
+            init: function() {
+
+                // Place initialization logic here
+                // You already have access to the DOM element and
+                // the options via the instance, e.g. this.element
+                // and this.settings
+                // you can add more functions like the one below and
+                // call them like the example below
+
+                this.pollingInterval = null;
+
+                this.pollNodes();
+            },
+
+            pollNodes: function () {
+                var self = this;
+                self.pollingInterval = setInterval(
+                    function () {
+                        var jqxhr = $.ajax({
+                            url: self.settings.nodesUrl,
+                            cache: false,
+                            context: self,
+                            dataType: "json"
+                        }).then(
+                            //On success
+                            self.processNodes,
+                            // On failure
+                            self.handleError
+                        );
+                    },
+                    self.settings.timeoutInterval
+                );
+            },
+
+            stopPolling: function () {
+                clearInterval(this.pollingInterval);
+            },
+
+            processNodes: function (nodesResponse) {
+                var self = this;
+                $.each(nodesResponse, function (index, value) {
+                     self.renderNode(value);
+                });
+            },
+
+            renderNode: function (node) {
+
```

Comment:
A coding style guide needs to be defined for the project and enforced through `eslint` or any similar linting tool for javascript